src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs

```diff
@@ -325,6 +325,10 @@ private void LogRequest(string methodName, string?[] providedParameters, Paramet
                     if (providedParameter.StartsWith('[') || providedParameter.StartsWith('{'))
                     {
                         executionParam = _serializer.Deserialize(new JsonTextReader(new StringReader(providedParameter)), paramType);
+                        if (executionParam is null && !IsNullableParameter(expectedParameter))
+                        {
+                            throw new JsonReaderException($"Cannot deserialize parameter `{providedParameter}` into type {paramType}");
```

Comment:
Shouldn't we just add `executionParameters.Add(Type.Missing)` ?

Reply:
I'm not sure, is an invalid parameter actually missing? From the users perspective, if we use the current approach, the result looks like:
```
{"jsonrpc":"2.0","error":{"code":-32602,"message":"Invalid params"},"id":1}
```

If we use `executionParameters.Add(Type.Missing)`, then the result looks like:
```
{"jsonrpc":"2.0","error":{"code":-32602,"message":"Parameter count mismatch.","data":"System.Reflection.TargetParameterCountException: Parameter count mismatch.
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Nethermind.JsonRpc.JsonRpcService.ExecuteAsync(JsonRpcRequest request, String methodName, ValueTuple`2 method, JsonRpcContext context) in /home/emlautarom1/Development/Nethermind/nethermind/src/Nethermind/Nethermind.JsonRpc/JsonRpcService.cs:line 186"},"id":1}
```

Reply:
Looking at this code: https://github.com/NethermindEth/nethermind/pull/6104/files/f5792fc8839aeea94d816b2266018a17591d2834#diff-bd27ffa6f72471764da77fa1d9e723ffb5eb275805fa6ee91380ee247d9d6549R295-R305

It should be `missing`. Leaking internal exception details here is also not great.

Reply:
It has its pro's and con's. I think it is fine for easier debugging user issues (For example, NullReferenceException on eth_createAccessList). But we shouldn't use exceptions on expected API errors.